clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -181,13 +182,28 @@ public static boolean validateRequiredResourceLabels(Map<String, String> metadat
         return validateResourceLabel(metadata, MetricsContext.NAMESPACE);
     }
 
-    public static CompressionType preferredCompressionType(List<CompressionType> acceptedCompressionTypes) {
-        if (acceptedCompressionTypes != null && !acceptedCompressionTypes.isEmpty()) {
-            // Broker is providing the compression types in order of preference. Grab the
-            // first one.
-            return acceptedCompressionTypes.get(0);
+    /**
+     * Determines the preferred compression type from broker-accepted types, avoiding unsupported ones.
+     * 
+     * @param acceptedCompressionTypes the list of compression types accepted by the broker in order 
+     *                                of preference, or null/empty if no compression is accepted
+     * @param unsupportedCompressionTypes the set of compression types that should be avoided due to 
+     *                                   missing libraries or previous failures (must not be null)
+     * @return the preferred compression type to use, or {@link CompressionType#NONE} if no acceptable
+     *         compression type is available
+     * @throws NullPointerException if unsupportedCompressionTypes is null
+     */
+    public static CompressionType preferredCompressionType(List<CompressionType> acceptedCompressionTypes, Set<CompressionType> unsupportedCompressionTypes) {
+        if (acceptedCompressionTypes == null) {
```

Comment:
According to the source code (https://github.com/apache/kafka/blob/9a2f202a1e99a807b6ba91bb3c4b9148089f0b02/clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java#L953), it appears the `acceptedCompressionTypes` could never be null.